drivers/SmartThings/matter-switch/src/aqara-cube/init.lua

```diff
@@ -126,53 +124,69 @@ local function configure_buttons(device)
   end
 end
 
-local function device_added(driver, device)
-  if device.network_type ~= device_lib.NETWORK_TYPE_CHILD then
-    local MS = get_reordered_endpoints(driver, device)
-    local main_endpoint = device.MATTER_DEFAULT_ENDPOINT
-    if #MS > 0 then
-      main_endpoint = MS[1] -- the endpoint matching to the non-child device
-      if MS[1] == 0 then main_endpoint = MS[2] end -- we shouldn't hit this, but just in case
-    end
-    device.log.debug("main button endpoint is "..main_endpoint)
-
-    -- At the moment, we're taking it for granted that all momentary switches only have 2 positions
-    local current_component_number = 1
-    local component_map = {}
-    for _, ep in ipairs(MS) do -- for each momentary switch endpoint (including main)
-      device.log.debug("Configuring endpoint "..ep)
-      -- build the mapping of endpoints to components
-      component_map[string.format("%d", current_component_number)] = ep
-      current_component_number = current_component_number + 1
+local function set_configure(driver, device)
+  local MS = get_reordered_endpoints(driver, device)
+  local main_endpoint = device.MATTER_DEFAULT_ENDPOINT
+  if #MS > 0 then
+    main_endpoint = MS[1] -- the endpoint matching to the non-child device
+    if MS[1] == 0 then main_endpoint = MS[2] end -- we shouldn't hit this, but just in case
+  end
+  device.log.debug("main button endpoint is "..main_endpoint)
+
+  -- At the moment, we're taking it for granted that all momentary switches only have 2 positions
+  local current_component_number = 1
+  local component_map = {}
+  for _, ep in ipairs(MS) do -- for each momentary switch endpoint (including main)
+    device.log.debug("Configuring endpoint "..ep)
+    -- build the mapping of endpoints to components
+    component_map[string.format("%d", current_component_number)] = ep
+    current_component_number = current_component_number + 1
+  end
+
+  device:set_field(COMPONENT_TO_ENDPOINT_MAP_BUTTON, component_map, {persist = true})
+  device:try_update_metadata({profile = "cube-t1-pro"})
+  configure_buttons(device)
+end
+
+local function device_init(driver, device)
+  if device.network_type == device_lib.NETWORK_TYPE_MATTER then
+    device:subscribe()
+    device:set_endpoint_to_component_fn(endpoint_to_component)
+
+    -- during driver update, only device_init is called, so configuration including profile change should be set here.
+    -- vid_0x115f_Cube is a label used in the unit test. This is used to avoid profile updates because the test framework
+    -- does not allow profile updates in the device_init stage. And it must be executed unconditionally in the user environment.
+    local name = string.format("%s", device.label)
+    if string.find(name, "vid_0x115f_Cube") == nil then
+      set_configure(driver, device)
```

Comment:
What are you seeing in the test environment that is forcing you to add this handling? Does expect_metadata_update not work for you?

Reply:
```suggestion
    -- when unit testing, call set_configure elsewhere
    if not device:get_field(TEST_CONFIGURE) then
      set_configure(driver, device)
   end
```
however, if we cannot find a solution to your unit test problem, I suggest writing something more like the above. It reuses the TEST_CONFIGURE variable, which keeps things neater.

Reply:
There was a problem in my test case file. I resolved the problem and applied your suggestion. Thanks~

Reply:
> What are you seeing in the test environment that is forcing you to add this handling? Does expect_metadata_update not work for you?

There is no problem calling `set_configure` in normal use cases. However, if there is a `metadata` update in the `init` during the unit test, an assertion occurs during the test case `init`. That's why I added it to avoid this situation during the unit test.